### DISCUSSION TITLE
Help, calling command in another command

### DISCUSSION THREAD

**Discussion by magic-thomas** [adonisjs/core#2073] [General]
Yes, I have to call child process command in command. 

I have two command 

call:invest -> child process
main:call  -> main process. 

in shell. 

```sh
adonis call:invest 
```

it works fine and produce infinite output . 

but for getting output from child process. 

```sh
adonis main:call 
```
-> it goest error. 

```sh 
child launch failed:  Error: spawn export NODE_ENV=test || set NODE_ENV=test && adonis call:invest ENOENT
    at Process.ChildProcess._handle.onexit (node:internal/child_process:276:19)
    at onErrorNT (node:internal/child_process:476:16)
    at processTicksAndRejections (node:internal/process/task_queues:80:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn export NODE_ENV=test || set NODE_ENV=test && adonis call:invest',
  path: 'export NODE_ENV=test || set NODE_ENV=test && adonis call:invest',
  spawnargs: []
}

```

my main:call coes are below 

```js
    const cmdStr ='export NODE_ENV=test || set NODE_ENV=test && adonis call:invest';
   //    const cmdStr ='adonis call:invest';  --> it goes  error also.
    const child = spawn( cmdStr);
```

How can I run command in another command ?

 p.s. 

I even tried with this codes 

```js
    const child = spawn( cmdStr, {
      stdio: 'inherit', // 'pipe', 'ignore' .. all goes error 
      shell: true
    }); 

```

## Package version
adonis 4.1  / puppeteer 

## Node.js and npm version
node : 15.4 
npm 6.14.10

**Comment by RomainLanz** [comment]
Hey @magic-thomas! 👋 

You can require `ace` and run the `ace.call()` method.

```js
const ace = require('@adonisjs/ace')

await ace.call('migration:run', {}, { silent: true })
```

**Reply by Agilulfo1820** [reply]
I have same problem as the OP, the only difference is that I need to call the command from inside a Task (cronjob).
Unfortunatly it always give me this error:
```
error: update:teams is not a registered command {"task":"update:teams"}
```

The command is registered, I can call it via cli with `adonis update:teams` or `node ace update:teams`.
